raiden/api/python.py

```diff
@@ -1194,6 +1208,58 @@ def get_token_action(self, token):
         token_data = self.raiden.wal.storage.query_token_action(token)
         return token_data
 
+    def create_invoice(self, data):
+
+        if data['already_coded_invoice'] is False:
+            chain = self.raiden.chain
+            private_key = chain.client.privkey.hex()
+            timestamp = get_utc_unix_time()
+            currency = data['currency_symbol']
+            fallback = None
+            amount = data['amount']
+            payment_hash = encrypt_string_with_sha256(random_string(20))
+            description = data['description']
+            description_hashed = None
+            expires = 3600
+            route = []
+            beneficiary = "0x" + data['partner_address'].hex()
+            token = "0x" + data['token_address'].hex()
+
+            options_args = OptionsArgs(timestamp,
+                                       currency,
+                                       fallback,
+                                       amount,
+                                       payment_hash,
+                                       description,
+                                       description_hashed,
+                                       expires,
+                                       route,
+                                       private_key,
+                                       beneficiary,
+                                       token)
+
+            lumino_invoice_obj = parse_options(options_args)
+
+            lumino_invoice_encoded = encode_invoice(lumino_invoice_obj, options_args.privkey)
+
+            expiration_date = get_utc_expiration_time(expires)
+
+        else:
+            payment_hash = data['payment_hash'].hex()
+            expiration_date = data['expiration_date']
+            lumino_invoice_encoded = data['encode']
+
+        lumino_invoice_persist = {"type": data['invoice_type'],
+                                  "status": data['invoice_status'],
+                                  "expiration_date": expiration_date,
+                                  "encode": lumino_invoice_encoded,
+                                  "payment_hash" : payment_hash}
+
+        # Save this information
+        self.raiden.wal.storage.write_invoice(lumino_invoice_persist)
```

Comment:
We should persist the entire invoice data and not just the encoding. If we need the encoded invoice we cal always encode it.

Reply:
ok

Reply:
So it would be saved now:

create table invoices
(
    identifier      INTEGER
        constraint invoices_pk
            primary key autoincrement,
    type            INTEGER,
    status          INTEGER,
    expiration_date TEXT,
    encode          TEXT,
    payment_hash    TEXT,
    secret          TEXT,
    currency        TEXT,
    amount          TEXT,
    description     TEXT,
    target_address  TEXT,
    token_address   TEXT,
    created_at      TEXT
);